core/src/main/java/org/apache/iceberg/FieldMetrics.java

```diff
@@ -30,15 +28,15 @@
   private final long valueCount;
   private final long nullValueCount;
   private final long nanValueCount;
-  private final ByteBuffer lowerBound;
-  private final ByteBuffer upperBound;
+  private final Object lowerBound;
+  private final Object upperBound;
 
   public FieldMetrics(int id,
                       long valueCount,
                       long nullValueCount,
                       long nanValueCount,
-                      ByteBuffer lowerBound,
-                      ByteBuffer upperBound) {
+                      Object lowerBound,
+                      Object upperBound) {
```

Comment:
If this is going to be the value instead of the serialized value, then the class should be parameterized by the value type: `FieldMetrics<T>` with `public T lowerBound()`.

Reply:
Yes, sorry I forgot to address that comment after attempting the alternative approach (mentioned in description). I have updated this PR to address this. And sorry for the delay!